python/test/state_representation/space/joint/test_joint_state.py

```diff
@@ -75,6 +76,383 @@ def test_truthiness(self):
+            res = positions - torques
+
+        with self.assertRaises(TypeError):
+            res = velocities - positions
+        with self.assertRaises(TypeError):
+            res = velocities - accelerations
+        with self.assertRaises(TypeError):
+            res = velocities - torques
+
+        with self.assertRaises(TypeError):
+            res = accelerations - positions
+        with self.assertRaises(TypeError):
+            res = accelerations - velocities
+        with self.assertRaises(TypeError):
+            res = accelerations - torques
+
+        with self.assertRaises(TypeError):
+            res = torques - positions
+        with self.assertRaises(TypeError):
+            res = torques - velocities
+        with self.assertRaises(TypeError):
+            res = torques - accelerations
+
+        state -= state
+        self.assertIsInstance(state, JointState)
+        state -= positions
+        self.assertIsInstance(state, JointState)
+        state -= velocities
+        self.assertIsInstance(state, JointState)
+        state -= accelerations
+        self.assertIsInstance(state, JointState)
+        state -= torques
+        self.assertIsInstance(state, JointState)
+
+        positions -= state
+        self.assertIsInstance(positions, JointPositions)
+        positions -= positions
+        self.assertIsInstance(positions, JointPositions)
+        with self.assertRaises(TypeError):
+            positions -= velocities
+        with self.assertRaises(TypeError):
+            positions -= accelerations
+        with self.assertRaises(TypeError):
+            positions -= torques
+
+        velocities -= state
+        self.assertIsInstance(velocities, JointVelocities)
+        velocities -= velocities
+        self.assertIsInstance(velocities, JointVelocities)
+        with self.assertRaises(TypeError):
+            velocities -= positions
+        with self.assertRaises(TypeError):
+            velocities -= accelerations
+        with self.assertRaises(TypeError):
+            velocities -= torques
+
+        accelerations -= state
+        self.assertIsInstance(accelerations, JointAccelerations)
+        accelerations -= accelerations
+        self.assertIsInstance(accelerations, JointAccelerations)
+        with self.assertRaises(TypeError):
+            accelerations -= positions
+        with self.assertRaises(TypeError):
+            accelerations -= velocities
+        with self.assertRaises(TypeError):
+            accelerations -= torques
+
+        torques -= state
+        self.assertIsInstance(torques, JointTorques)
+        torques -= torques
+        self.assertIsInstance(torques, JointTorques)
+        with self.assertRaises(TypeError):
+            torques -= positions
+        with self.assertRaises(TypeError):
+            torques -= velocities
+        with self.assertRaises(TypeError):
+            torques -= accelerations
+
+    def test_multiplication_operators(self):
```

Comment:
Could you 

What happens if you try to do the time-based operations on classes that cannot be integrated / differentiated? I would expect you get a TypeError automatically, but worse would be if it somehow does something valid, in which case it would need explicit bindings to throw an exception.

Either way it would be nice to see the test cases for all variants e.g.
```python
with self.assertRaises(TypeError):
    timedelta(seconds=1) * state
with self.assertRaises(TypeError):
    state / timedelta(seconds=1)
```